File: app/src/main/java/com/udacity/popularmovies/network/TheMovieDBService.java
package com.udacity.popularmovies.network;

import com.udacity.popularmovies.model.MovieReview;
import com.udacity.popularmovies.model.MovieTrailer;
import com.udacity.popularmovies.model.Movies;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;


// Interface for Retrofit Implementation
public interface TheMovieDBService {

    @GET("/3/discover/movie")
    Call  <Movies> getData(
            @Query("api_key") String apiKey,
            @Query("sort_by") String sortOrder);

    @GET("/3/movie/{id}/videos")
    Call  <MovieTrailer> getTrailer(
            @Path("id") int id,
            @Query("api_key") String apiKey);

    @GET("/3/movie/{id}/reviews")
    Call  <MovieReview> getReview(
            @Path("id") int id,
            @Query("api_key") String apiKey);



}

File: app/src/main/java/com/udacity/popularmovies/network/TheMovieDBService.kt
package com.udacity.popularmovies.network

import com.udacity.popularmovies.model.MovieReview
import com.udacity.popularmovies.model.MovieTrailer
import com.udacity.popularmovies.model.Movies
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

// Interface for Retrofit Implementation
interface TheMovieDBService {
    @GET("/3/discover/movie")
    suspend fun getData(
            @Query("api_key") apiKey: String?,
            @Query("sort_by") sortOrder: String?): Movies?
            //@Query("sort_by") sortOrder: String?): Call<Movies?>?


    @GET("/3/movie/{id}/videos")
    suspend fun getTrailer(
            @Path("id") id: Int,
            @Query("api_key") apiKey: String?): Call<MovieTrailer?>?

    @GET("/3/movie/{id}/reviews")
    suspend fun getReview(
            @Path("id") id: Int,
            @Query("api_key") apiKey: String?): Call<MovieReview?>?
}